src/imitation/data/rollout.py

```diff
@@ -107,17 +108,22 @@ def finish_trajectory(
         for part_dict in part_dicts:
             for k, array in part_dict.items():
                 out_dict_unstacked[k].append(array)
-        out_dict_stacked = {
-            k: np.stack(arr_list, axis=0) for k, arr_list in out_dict_unstacked.items()
-        }
-        traj = types.TrajectoryWithRew(**out_dict_stacked, terminal=terminal)
-        assert traj.rews.shape[0] == traj.acts.shape[0] == traj.obs.shape[0] - 1
+
+        # TODO: what about infos? Does this actually handle them well?
+        traj = types.TrajectoryWithRew(
+            obs=types.stack_maybe_dictobs(out_dict_unstacked["obs"]),
+            acts=np.stack(out_dict_unstacked["acts"], axis=0),
+            infos=np.stack(out_dict_unstacked["infos"], axis=0),  # TODO: confused
```

Comment:
I think it forms a numpy object-dtype array. It's a bit silly/unnecessary that this is a NumPy array rather than just a list but AFAIK it works OK.